packages/backend-function/src/resolve_secret_banner.ts

```diff
@@ -7,31 +7,58 @@ import { SSM } from '@aws-sdk/client-ssm';
 export const internalAmplifyFunctionBannerResolveSecrets = async (
   client?: SSM
 ) => {
-  if (!client) {
-    client = new SSM();
-  }
-
   const envPathObj: Record<string, string> = JSON.parse(
     process.env.AMPLIFY_SECRET_PATHS ?? '{}'
   );
+  const sharedEnvPathObj: Record<string, string> = JSON.parse(
+    process.env.AMPLIFY_SHARED_SECRET_PATHS ?? '{}'
+  );
   const paths = Object.values(envPathObj);
+  const sharedPaths = Object.values(sharedEnvPathObj);
 
-  if (paths.length === 0) {
+  if (paths.length === 0 && sharedPaths.length === 0) {
     return;
   }
 
-  const response = await client.getParameters({
-    Names: paths,
-    WithDecryption: true,
-  });
-  if (response.Parameters && response.Parameters?.length > 0) {
-    for (const parameter of response.Parameters) {
-      for (const [env, path] of Object.entries(envPathObj)) {
-        if (parameter.Name === path) {
-          process.env[env] = parameter.Value;
+  const resolveSecrets = async (
+    paths: string[],
+    envPathObject: Record<string, string>
+  ) => {
+    if (!client) {
+      client = new SSM();
+    }
+
+    const response = await client.getParameters({
+      Names: paths,
+      WithDecryption: true,
+    });
+
+    if (response.Parameters && response.Parameters.length > 0) {
+      for (const parameter of response.Parameters) {
+        for (const [env, path] of Object.entries(envPathObject)) {
+          if (parameter.Name === path) {
+            process.env[env] = parameter.Value;
+          }
         }
       }
     }
+
+    return response;
+  };
+
+  const response = await resolveSecrets(paths, envPathObj);
+
+  if (response.InvalidParameters && response.InvalidParameters.length > 0) {
+    const pathsToResolve: string[] = [];
+    for (const invalidParameter of response.InvalidParameters) {
+      const env = Object.keys(envPathObj).find(
```

Comment:
It's starting to look like we should change how we are storing this metadata. What if we stored the following instead:
```ts
{
  'ssm/branch/path/for/secret1': {
    name: 'ENV_VAR_NAME1',
    sharedPath: 'ssm/shared/path/for/secret1'
  },
  'ssm/branch/path/for/secret2': {
    name: 'ENV_VAR_NAME2',
    sharedPath: 'ssm/shared/path/for/secret2'
  },
  etc
}
```

Then this code becomes something like
```ts
const response = await resolveSecrets(Object.keys(envPathObject));

const sharedPaths = (response.InvalidParameters || [])
  .map(invalidParam => envPathObject[invalidParam]?.sharedPath)
  .filter(sharedParam => !!sharedParam);

await resolveSecrets(sharedPaths)
```

This would also do away with the need for the additional environment variable in the function config

Reply:
Oh I like this structure for the metadata, yeah we can go with this.

Reply:
Curious why you went for branch path as the parent key instead of as a sibling of `sharedPath`? Is that just for the simplicity for the initial response in this line?
```ts
const response = await resolveSecrets(Object.keys(envPathObject));
```

Reply:
I don't have a strong opinion for one or the other, went with your suggested approach